pkg/local_object_storage/blobstor/fstree/head.go

```diff
@@ -0,0 +1,203 @@
+package fstree
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"os"
+
+	"github.com/klauspost/compress/zstd"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/util/logicerr"
+	apistatus "github.com/nspcc-dev/neofs-sdk-go/client/status"
+	objectSDK "github.com/nspcc-dev/neofs-sdk-go/object"
+	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
+	"github.com/nspcc-dev/neofs-sdk-go/proto/object"
+	"github.com/nspcc-dev/neofs-sdk-go/proto/refs"
+	"google.golang.org/protobuf/encoding/protowire"
+	"google.golang.org/protobuf/proto"
+)
+
+const (
+	_ = iota
+	fieldObjectID
+	fieldObjectSignature
+	fieldObjectHeader
+	fieldObjectPayload
+)
+
+// Head returns an object's header from the storage by address without reading the full payload.
+func (t *FSTree) Head(addr oid.Address) (*objectSDK.Object, error) {
+	p := t.treePath(addr)
+
+	f, err := os.Open(p)
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return nil, logicerr.Wrap(apistatus.ErrObjectNotFound)
+		}
+		return nil, fmt.Errorf("read file %q: %w", p, err)
+	}
+	defer f.Close()
+
+	obj, err := t.extractHeaderOnly(addr.Object(), f)
+	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) {
+			return nil, logicerr.Wrap(apistatus.ErrObjectNotFound)
+		}
+		return nil, fmt.Errorf("extract object header from %q: %w", p, err)
+	}
+
+	return obj, nil
+}
+
+// extractHeaderOnly reads the header of an object from a file.
+func (t *FSTree) extractHeaderOnly(id oid.ID, f *os.File) (*objectSDK.Object, error) {
+	buf := make([]byte, 2*objectSDK.MaxHeaderLen)
```

Comment:
can be just `buf := make([]byte, objectSDK.MaxHeaderLen, 2*objectSDK.MaxHeaderLen)` and no need to reslice below